### PR TITLE
bpo-34793: Drop old-style context managers in asyncio.locks

### PR DESCRIPTION
**Pull request by asvetlov**
https://bugs.python.org/issue34793

### CODE REVIEW

File: Doc/whatsnew/3.9.rst [hunk 1]
```diff
@@ -337,6 +337,11 @@ Removed
   of :pep:`442`. Patch by Joannah Nanjekye.
   (Contributed by Joannah Nanjekye in :issue:`15088`)
 
+* ``with (await asyncio.lock):`` and ``with (yield from asyncio.lock):`` statements are
+  not longer supported, use ``async with lock`` instead.  The same is correct for
+  ``asyncio.Condition`` and ``asyncio.Semaphore``.
+  (Contributed by Andrew Svetlov in :issue:`34793`.)
```
Comment (1st1) [inline Doc/whatsnew/3.9.rst:476]: Please also replace the deprecation warning with changedversion in https://docs.python.org/3/library/asyncio-sync.html#boundedsemaphore

File: Lib/asyncio/locks.py [hunk 2]
```diff
@@ -3,96 +3,13 @@
...
-        try:
-            self._lock.release()
-        finally:
-            self._lock = None  # Crudely prevent reuse.
 
 
 class _ContextManagerMixin:
-    def __enter__(self):
-        raise RuntimeError(
-            '"yield from" should be used as context manager expression')
```
Comment (1st1) [inline Lib/asyncio/locks.py:52]: Should we keep this? We can change the error message to say that `async with should be used...`

File: Doc/whatsnew/3.9.rst (hunk 1 above)
Comment (asvetlov) [reply Doc/whatsnew/3.9.rst:476]: Done

File: Lib/asyncio/locks.py (hunk 2 above)
Comment (asvetlov) [reply Lib/asyncio/locks.py:52]: I think the motivation was emphasizing that context managers were supported but in a slightly unusual way: `with (yield from lock):`; bare `with lock:` was forbidden.

After switching to `async with lock:` everything becomes clear: `with` is not supported in any form for the sake of `async with`. The error message is standard and well-known: `AttributeError: __enter__`.  I think we shouldn't replace the default behavior.

Maybe the standard text can be improved globally, but it is another story, I guess.

**Pull request merged**